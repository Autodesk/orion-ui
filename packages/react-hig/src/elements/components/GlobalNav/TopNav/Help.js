/**
Copyright 2016 Autodesk,Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

 */
import HIGElement from '../../../HIGElement';
import * as PropTypes from 'prop-types';
import createComponent from '../../../../adapters/createComponent';

export class Help extends HIGElement {
  commitUpdate(updatePayload, oldProps, newProps) {
    const mapping = {
      title: 'setTitle',
      link: 'setLink'
    };

    for (let i = 0; i < updatePayload.length; i += 2) {
      const propKey = updatePayload[i];
      const propValue = updatePayload[i + 1];

      if (mapping[propKey]) {
        this.hig[mapping[propKey]](propValue);
      } else {
        this.commitPropChange(propKey, propValue);
      }
    }
  }
}

const HelpComponent = createComponent(Help);

HelpComponent.propTypes = {
  title: PropTypes.string,
  link: PropTypes.string
};

HelpComponent.__docgenInfo = {
  props: {
    title: {
      description: 'sets the title of a Help shortcut'
    },

    link: {
      description: 'sets the link of a Help shortcut'
    }
  }
};

export default HelpComponent;
