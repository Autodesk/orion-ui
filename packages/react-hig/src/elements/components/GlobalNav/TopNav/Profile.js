/**
Copyright 2016 Autodesk,Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

*/

import * as PropTypes from 'prop-types';
import HIGElement from '../../../HIGElement';
import createComponent from '../../../../adapters/createComponent';

export class Profile extends HIGElement {
  componentDidMount() {
    if (this.initialProps.open === true) {
      this.hig.open();
    }
  }

  commitUpdate(updatePayload, oldProps, newProp) {
    const mapping = {
      image: 'setImage',
      name: 'setName',
      email: 'setEmail',
      profileSettingsLabel: 'setProfileSettingsLabel',
      signOutLabel: 'setSignOutLabel',
      profileSettingsLink: 'setProfileSettingsLink'
    };

    const openIndex = updatePayload.indexOf('open');
    if (openIndex >= 0) {
      const [openKey, openSetting] = updatePayload.splice(openIndex, 2);
      if (openKey) {
        if (openSetting === true) {
          this.hig.open();
        } else {
          this.hig.close();
        }
      }
    }

    this.commitUpdateWithMapping(updatePayload, mapping);
  }
}

const ProfileComponent = createComponent(Profile);

ProfileComponent.propTypes = {
  image: PropTypes.string,
  name: PropTypes.string,
  email: PropTypes.string,
  signOutLabel: PropTypes.string,
  profileSettingsLabel: PropTypes.string,
  profileSettingsLink: PropTypes.string,
  onSignOutClick: PropTypes.func,
  onProfileImageClick: PropTypes.func,
  onProfileClickOutside: PropTypes.func
};

ProfileComponent.__docgenInfo = {
  props: {
    open: {
      description: '{bool} sets whether flyout is open or closed'
    },
    name: {
      description: 'sets {String} display name'
    },
    email: {
      description: 'sets {String} email'
    },
    image: {
      description: 'sets {String} image - url to svg or other image object'
    },
    signOutLabel: {
      description: 'sets {String} label for sign out button'
    },
    profileSettingsLabel: {
      description: 'sets {String} label for settings button'
    },
    profileSettingsLink: {
      description: 'sets {String} url to settings page'
    },
    onSignOutClick: {
      description: '{function} Triggered when user clicks sign out button'
    },
    onProfileImageClick: {
      description: '{function} triggers when you click the profile'
    },
    onProfileClickOutside: {
      description: '{function} triggers when you click anywhere but the profile flyout'
    }
  }
};

export default ProfileComponent;
