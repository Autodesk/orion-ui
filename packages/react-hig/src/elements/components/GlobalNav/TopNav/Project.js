/**
Copyright 2016 Autodesk,Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

*/

import * as PropTypes from 'prop-types';
import HIGElement from '../../../HIGElement';
import createComponent from '../../../../adapters/createComponent';

export class Project extends HIGElement {
  constructor(HIGConstructor, initialProps) {
    super(HIGConstructor, initialProps);

    this.props = { ...initialProps };
    this.callOnActivateCallback = this.callOnActivateCallback.bind(this);
  }

  componentDidMount() {
    this.hig.onClick(this.callOnActivateCallback);
  }

  commitUpdate(updatePayload, oldProps, newProp) {
    this.props = { ...this.props, ...updatePayload };

    this.processUpdateProps(updatePayload)
      .mapToHIGFunctions({
        image: 'setImage',
        label: 'setLabel'
      })
      .mapToHIGEventListeners(['onClick'])
      .handle('active', value => {
        if (value) {
          this.hig.activate();
          this.onActivate(this);
        } else {
          this.hig.deactivate();
        }
      });
  }

  onActivate(callback) {
    this._onActivate = callback;
  }

  callOnActivateCallback() {
    if (!this._onActivate) {
      return
    }

    this._onActivate(this);
  }
}

const ProjectComponent = createComponent(Project);

ProjectComponent.propTypes = {
  image: PropTypes.string,
  label: PropTypes.string,
  activate: PropTypes.func,
  deactivate: PropTypes.func,
  onClick: PropTypes.func
};

ProjectComponent.__docgenInfo = {
  props: {
    label: {
      description: 'sets {String} the label displayed for a project in project/account switcher'
    },
    image: {
      description: 'sets {String} the image displayed for a project in project/account switcher'
    },
    activate: {
      description: '{func} activates the project'
    },
    deactivate: {
      description: '{func} deactivates the project'
    },
    onClick: {
      description: '{func} calls the provided callback when user clicks on the project'
    }
  }
};

export default ProjectComponent;
