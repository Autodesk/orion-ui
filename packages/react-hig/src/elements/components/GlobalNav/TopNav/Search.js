/* Copyright 2016 Autodesk,Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

 */

import HIGElement from '../../../HIGElement';
import * as PropTypes from 'prop-types';
import createComponent from '../../../../adapters/createComponent';

export class Search extends HIGElement {
  componentDidMount() {
    this.hig.onInput(this.topNavSearchOnInput.bind(this));
    this.hig.onClearIconClick(this.topNavClearInput.bind(this));
  }

  commitUpdate(updatePayload, oldProps, newProps) {
    const mapping = {
      placeholder: 'setPlaceholder',
    };
    super.commitUpdateWithMapping(updatePayload, mapping);
  }

  topNavSearchOnInput(event) {
    this.hig.showClearIcon();
    this.hig.setQuery(event.target.value);
  };

  topNavClearInput(event) {
    this.hig.setQuery("");
    this.hig.hideClearIcon();
  };

}

const SearchComponent = createComponent(Search);

SearchComponent.propTypes = {
  placeholder: PropTypes.string,
  query: PropTypes.string,
  showClearIcon: PropTypes.func,
  hideClearIcon: PropTypes.func,
  onClearIconClick: PropTypes.func,
  onInput: PropTypes.func,
  onFocusIn: PropTypes.func,
  onFocusOut: PropTypes.func
};

SearchComponent.__docgenInfo = {
  props: {
    setPlaceholder: { description: "{String} sets the placeholder text" },
    setQuery: { description: "{String} sets the search query text" },
    showClearIcon: { description: "{function} shows the clear input icon" },
    hideClearIcon: { description: "{function} hides the clear input icon" },
    onClearIconClick: { description: "{function} Triggers when user clicks clear icon" },
    onInput: { description: "{function} Triggers when user enters text into search field"},
    onFocusIn: { description: "{function} Triggers when input has focus"},
    onFocusOut: { description: "{function} Triggers when input loses focus"}
  }
};
export default SearchComponent;
