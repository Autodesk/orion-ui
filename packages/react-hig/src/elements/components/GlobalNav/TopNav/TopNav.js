/**
Copyright 2016 Autodesk,Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

*/
//import * as HIG from 'hig.web';
import * as PropTypes from 'prop-types';

import HIGElement from '../../../HIGElement';
import HIGChildValidator from '../../../HIGChildValidator';
import createComponent from '../../../../adapters/createComponent';

import HIGNodeList from '../../../HIGNodeList';
import ProjectAccountSwitcherComponent, {
  ProjectAccountSwitcherAdapter
} from '../../../../adapters/ProjectAccountSwitcherAdapter';
import ProfileComponent, {
  ProfileAdapter
} from '../../../../adapters/ProfileAdapter';
import ShortcutComponent, { Shortcut } from './Shortcut';
import HelpComponent, { Help } from './Help';
import SearchComponent, { Search } from './Search';

export class TopNav extends HIGElement {
  constructor(HIGConstructor, initialProps) {
    super(HIGConstructor, initialProps);

    this.shortcuts = new HIGNodeList({
      type: Shortcut,
      HIGConstructor: this.hig.partials.Shortcut,
      onAdd: (instance, beforeInstance) => {
        this.hig.addShortcut(instance, beforeInstance);
      }
    });
  }

  componentDidMount() {
    // Add any children
    if (this.profile) {
      this.hig.addProfile(this.profile.hig);
      this.profile.mount();
    }

    if (this.projectAccountSwitcher) {
      this.hig.addProjectAccountSwitcher(this.projectAccountSwitcher.hig);
      this.projectAccountSwitcher.mount();
    }

    this.shortcuts.componentDidMount();

    if (this.help) {
      this.hig.addHelp(this.help.hig);
      this.help.mount();
    }

    if (this.search) {
      this.hig.addSearch(this.search.hig);
      this.search.mount();
    }
  }

  commitUpdate(updatePayload, oldProps, newProp) {
    this.processUpdateProps(updatePayload).mapToHIGFunctions({
      logo: 'setLogo',
      logoLink: 'setLogoLink'
    });
  }

  createElement(ElementConstructor, props) {
    switch (ElementConstructor) {
      case ProfileAdapter:
        return new ProfileAdapter(this.hig.partials.Profile, props);
      case ProjectAccountSwitcherAdapter:
        return new ProjectAccountSwitcherAdapter(
          this.hig.partials.ProjectAccountSwitcher,
          props
        );
      case Help:
        return new Help(this.hig.partials.Help, props);
      case Shortcut:
        return this.shortcuts.createElement(ElementConstructor, props);
      case Search:
        return new Search(this.hig.partials.Search, props);
      default:
        throw new Error(`Unknown type ${ElementConstructor.name}`);
    }
  }

  appendChild(instance, beforeChild = {}) {
    this.requireSingleInstance(instance);
    this.checkValidChild(instance);

    if (instance instanceof Shortcut) {
      this.shortcuts.insertBefore(instance);
    } else {
      this[this.getPropertyNameFor(instance)] = instance;
      if (this.mounted) {
        this.hig[this.getFunctionNameFor(instance)](instance.hig);
        instance.mount();
      }
    }
  }

  insertBefore(instance) {
    this.appendChild(instance);
  }

  removeChild(instance) {
    if (instance instanceof ProfileAdapter) {
      this.profile = null;
    }

    instance.unmount();
  }

  requireSingleInstance(instance) {
    const requiredSingle = [
      'Profile',
      'ProjectAccountSwitcher',
      'Help',
      'Search'
    ];
    super.requireSingleInstance(instance, requiredSingle);
  }

  checkValidChild(instance) {
    const validInstances = [
      ProfileAdapter,
      ProjectAccountSwitcherAdapter,
      Shortcut,
      Help,
      Search
    ];
    if (!validInstances.includes(instance.constructor)) {
      throw new Error(
        `${instance.constructor.name} is not a valid child element of this parent.`
      );
    }
  }

  getPropertyNameFor(instance) {
    if (instance instanceof ProfileAdapter) {
      return 'profile';
    }
    if (instance instanceof ProjectAccountSwitcherAdapter) {
      return 'projectAccountSwitcher';
    }
    if (instance instanceof Shortcut) {
      return 'shortcut';
    }
    if (instance instanceof Help) {
      return 'help';
    }
    if (instance instanceof Search) {
      return 'search';
    }
    return null;
  }

  getFunctionNameFor(instance) {
    return 'add' + instance.constructor.name;
  }

  onHamburgerClick(callback) {
    this.hig.onHamburgerClick(callback);
  }
}

const TopNavComponent = createComponent(TopNav);

TopNavComponent.propTypes = {
  logo: PropTypes.string,
  logoLink: PropTypes.string,
  onHamburgerClick: PropTypes.func,
  addProfile: PropTypes.func,
  addProjectAccountSwitcher: PropTypes.func,
  addSearch: PropTypes.func,
  children: HIGChildValidator([
    ProfileComponent,
    ProjectAccountSwitcherComponent,
    ShortcutComponent,
    HelpComponent,
    SearchComponent
  ])
};

TopNavComponent.__docgenInfo = {
  props: {
    logo: {
      description: 'sets the logo'
    },

    logoLink: {
      description: 'sets the link of the logo'
    },

    onHamburgerClick: {
      description: 'triggered when hamburger icon is clicked'
    },

    addProfile: {
      description: 'adds Profile to the top nav'
    },

    addProjectAccountSwitcher: {
      description: 'Pass in an instance of a ProjectAccountSwitcher partial to mount it to the TopNav'
    },

    addSearch: {
      description: 'Pass in an instance of a topNavSearch partial to mount to TopNav'
    }
  }
};

TopNavComponent.Profile = ProfileComponent;
TopNavComponent.Shortcut = ShortcutComponent;
TopNavComponent.Help = HelpComponent;
TopNavComponent.ProjectAccountSwitcher = ProjectAccountSwitcherComponent;
TopNavComponent.Search = SearchComponent;

export default TopNavComponent;
