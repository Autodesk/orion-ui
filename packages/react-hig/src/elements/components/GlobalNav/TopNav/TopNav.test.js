/**
Copyright 2016 Autodesk,Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

*/

import { mount } from 'enzyme';
import * as HIG from 'hig.web';
import React from 'react';

import GlobalNav from './../GlobalNav';
import TopNav from './TopNav';
import ProfileComponent, {
  ProfileAdapter
} from '../../../../adapters/ProfileAdapter';
import Search from './Search';

const Context = props => {
  return (
    <GlobalNav>
      <TopNav>
        <Search
          placeholder={props.placeholder}
          query={props.query}
          onInput={props.onInput}
          onFocusIn={props.onFocusIn}
          onFocusOut={props.onFocusOut}
        />
      </TopNav>
    </GlobalNav>
  );
};

describe('<TopNav>', () => {
  function createHigNav() {
    const domContainer = document.createElement('div');
    const higNav = new HIG.GlobalNav();
    higNav.mount(domContainer);

    return { higNav, domContainer };
  }

  function createTopNav() {
    const { higNav, domContainer } = createHigNav();
    const topNav = new higNav.partials.TopNav({ ...defaults });
    higNav.addTopNav(topNav);
    return { topNav, higNav, domContainer };
  }

  // Create the GlobalNav context for the TopNav to be attached to
  let defaults = {
    logo: '../../../bim-logo.png',
    logoLink: 'http://www.autodesk.com'
  };
  let profileDefaults = { image: '../../../bim-logo.png' };

  it('renders a topnav', () => {
    const reactContainer = document.createElement('div');

    mount(
      <GlobalNav>
        <TopNav {...defaults}>
          <ProfileComponent {...profileDefaults} />
        </TopNav>
      </GlobalNav>,
      { attachTo: reactContainer }
    );
    expect(reactContainer.firstElementChild.outerHTML).toMatchSnapshot();
  });

  it('can render a Profile', () => {
    const { topNav, higNav, domContainer } = createTopNav();

    const profile = new topNav.partials.Profile({ ...profileDefaults });
    topNav.addProfile(profile);

    const reactContainer = document.createElement('div');
    mount(
      <GlobalNav>
        <GlobalNav.TopNav {...defaults}>
          <ProfileComponent {...profileDefaults} />
        </GlobalNav.TopNav>
      </GlobalNav>,
      { attachTo: reactContainer }
    );

    expect(reactContainer.firstElementChild.outerHTML).toEqual(
      domContainer.firstElementChild.outerHTML
    );
  });

  function anyHandler() {
    return true;
  }

  it('can render Search programmatically', () => {
    const { topNav, higNav, domContainer } = createTopNav();
    const props = {
      placeholder: 'enter some text',
      query: 'foobar',
      onInput: anyHandler,
      onFocusIn: anyHandler,
      onFocusOut: anyHandler
    };
    const tnsearch = new topNav.partials.Search(props);
    topNav.addSearch(tnsearch);

    expect(domContainer.firstChild.outerHTML).toMatchSnapshot();
    const elems = domContainer.getElementsByClassName(
      'hig__global-nav__top-nav__search__inputholder__input'
    );
    expect(elems.length).toEqual(1);
  });

  it('can render Search through React-like components', () => {
    const reactContainer = document.createElement('div');
    const props = {
      placeholder: 'enter some text',
      query: 'foobar',
      onInput: anyHandler,
      onFocusIn: anyHandler,
      onFocusOut: anyHandler
    };

    mount(Context(props), { attachTo: reactContainer });
    expect(reactContainer.firstChild.outerHTML).toMatchSnapshot();

    const elems = reactContainer.getElementsByClassName(
      'hig__global-nav__top-nav__search__inputholder__input'
    );
    expect(elems.length).toEqual(1);
  });
});
