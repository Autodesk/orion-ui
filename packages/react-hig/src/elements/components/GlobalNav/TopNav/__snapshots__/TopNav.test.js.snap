// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopNav> can render Search programmatically 1`] = `
"<div class=\\"hig__global-nav\\">
    <!--SIDENAV-->
    <div class=\\"hig__global-nav__container\\">
        <div class=\\"hig__global-nav__top-nav\\">
    <div class=\\"hig__global-nav__top-nav__group hig__global-nav__top-nav__group--left\\">
        <div class=\\"hig__global-nav__top-nav__hamburger\\">
            <div class=\\"hig__global-nav__top-nav__hamburger__hamburgericon\\"><div class=\\"hig__icon\\"><svg width=\\"30\\" height=\\"30\\" viewBox=\\"0 0 30 30\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"hamburger\\"><g transform=\\"translate(-92 -185)\\" fill=\\"none\\" fill-rule=\\"evenodd\\"><rect x=\\"92\\" y=\\"185\\" width=\\"30\\" height=\\"30\\"></rect><path d=\\"M100,195 L115,195 L115,196 L100,196 L100,195 Z M100,200 L115,200 L115,201 L100,201 L100,200 Z M100,205 L115,205 L115,206 L100,206 L100,205 Z\\" fill=\\"#34495E\\"></path></g></svg></div></div>
            <div class=\\"hig__global-nav__top-nav__hamburger__closeicon\\"><div class=\\"hig__icon\\"><svg width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"close-hamburger\\"><title>hig_icon_xclose</title><g fill=\\"#353636\\" fill-rule=\\"evenodd\\"><rect transform=\\"rotate(45 12 12)\\" x=\\"4\\" y=\\"11\\" width=\\"16\\" height=\\"2\\"></rect><rect transform=\\"rotate(-45 12 12)\\" x=\\"4\\" y=\\"11\\" width=\\"16\\" height=\\"2\\"></rect></g></svg></div></div>
        </div>
        <div class=\\"hig__global-nav__top-nav__logo\\">
            <a href=\\"http://www.autodesk.com\\">
                <img src=\\"../../../bim-logo.png\\">
            </a>
        </div>
    </div>
    <div class=\\"hig__global-nav__top-nav__search\\">
    <div class=\\"hig__global-nav__top-nav__search__icon\\"><div class=\\"hig__icon\\"><svg width=\\"22\\" height=\\"22\\" viewBox=\\"0 0 22 22\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"search\\"><g transform=\\"translate(-1 -1)\\" fill=\\"none\\" fill-rule=\\"evenodd\\"><rect fill=\\"#353636\\" transform=\\"rotate(-45 19.196 18.891)\\" x=\\"18.195\\" y=\\"13.891\\" width=\\"2.001\\" height=\\"10\\" rx=\\"1\\"></rect><circle stroke=\\"#353636\\" stroke-width=\\"1.5\\" cx=\\"10\\" cy=\\"10\\" r=\\"8\\"></circle></g></svg></div></div>
    <div class=\\"hig__global-nav__top-nav__search__inputholder\\"><input class=\\"hig__global-nav__top-nav__search__inputholder__input\\" type=\\"text\\" placeholder=\\"enter some text\\" value=\\"foobar\\"></div>
    <div class=\\"hig__global-nav__top-nav__search__clear\\"><div class=\\"hig__icon\\"><svg width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"close-small\\"><title>hig/icon/close-small</title><g fill=\\"none\\" fill-rule=\\"evenodd\\"><circle stroke=\\"#353636\\" cx=\\"12\\" cy=\\"12\\" r=\\"8\\"></circle><polygon fill=\\"#353636\\" transform=\\"rotate(45 12 12)\\" points=\\"16.5 11.4518414 12.5847458 11.4518414 12.5847458 7.5 11.4152542 7.5 11.4152542 11.4518414 7.5 11.4518414 7.5 12.5226629 11.4152542 12.5226629 11.4152542 16.5 12.5847458 16.5 12.5847458 12.5226629 16.5 12.5226629\\"></polygon></g></svg></div></div>
</div><!--SEARCH-->
    <div class=\\"hig__global-nav__top-nav__spacer\\"></div>
    <!--PROJECT_ACCOUNT_SWITCHER-->
    <div class=\\"hig__global-nav__top-nav__group hig__global-nav__top-nav__group--right\\">
    <!--SHORTCUT-->
    </div>
    <!--PROFILE-->
</div><!--TOPNAV-->
        <!--SUBNAV-->
        <div class=\\"hig__global-nav__slot\\">
            <!--SLOT-->
        </div>
    </div>
</div>"
`;

exports[`<TopNav> can render Search through React-like components 1`] = `
"<div class=\\"hig__global-nav\\">
    <!--SIDENAV-->
    <div class=\\"hig__global-nav__container\\">
        <div class=\\"hig__global-nav__top-nav\\">
    <div class=\\"hig__global-nav__top-nav__group hig__global-nav__top-nav__group--left\\">
        <div class=\\"hig__global-nav__top-nav__hamburger\\">
            <div class=\\"hig__global-nav__top-nav__hamburger__hamburgericon\\"><div class=\\"hig__icon\\"><svg width=\\"30\\" height=\\"30\\" viewBox=\\"0 0 30 30\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"hamburger\\"><g transform=\\"translate(-92 -185)\\" fill=\\"none\\" fill-rule=\\"evenodd\\"><rect x=\\"92\\" y=\\"185\\" width=\\"30\\" height=\\"30\\"></rect><path d=\\"M100,195 L115,195 L115,196 L100,196 L100,195 Z M100,200 L115,200 L115,201 L100,201 L100,200 Z M100,205 L115,205 L115,206 L100,206 L100,205 Z\\" fill=\\"#34495E\\"></path></g></svg></div></div>
            <div class=\\"hig__global-nav__top-nav__hamburger__closeicon\\"><div class=\\"hig__icon\\"><svg width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"close-hamburger\\"><title>hig_icon_xclose</title><g fill=\\"#353636\\" fill-rule=\\"evenodd\\"><rect transform=\\"rotate(45 12 12)\\" x=\\"4\\" y=\\"11\\" width=\\"16\\" height=\\"2\\"></rect><rect transform=\\"rotate(-45 12 12)\\" x=\\"4\\" y=\\"11\\" width=\\"16\\" height=\\"2\\"></rect></g></svg></div></div>
        </div>
        <div class=\\"hig__global-nav__top-nav__logo\\">
            <a href=\\"#\\">
                <img src=\\"link\\">
            </a>
        </div>
    </div>
    <div class=\\"hig__global-nav__top-nav__search\\">
    <div class=\\"hig__global-nav__top-nav__search__icon\\"><div class=\\"hig__icon\\"><svg width=\\"22\\" height=\\"22\\" viewBox=\\"0 0 22 22\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"search\\"><g transform=\\"translate(-1 -1)\\" fill=\\"none\\" fill-rule=\\"evenodd\\"><rect fill=\\"#353636\\" transform=\\"rotate(-45 19.196 18.891)\\" x=\\"18.195\\" y=\\"13.891\\" width=\\"2.001\\" height=\\"10\\" rx=\\"1\\"></rect><circle stroke=\\"#353636\\" stroke-width=\\"1.5\\" cx=\\"10\\" cy=\\"10\\" r=\\"8\\"></circle></g></svg></div></div>
    <div class=\\"hig__global-nav__top-nav__search__inputholder\\"><input class=\\"hig__global-nav__top-nav__search__inputholder__input\\" type=\\"text\\" placeholder=\\"enter some text\\" value=\\"foobar\\"></div>
    <div class=\\"hig__global-nav__top-nav__search__clear\\"><div class=\\"hig__icon\\"><svg width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"close-small\\"><title>hig/icon/close-small</title><g fill=\\"none\\" fill-rule=\\"evenodd\\"><circle stroke=\\"#353636\\" cx=\\"12\\" cy=\\"12\\" r=\\"8\\"></circle><polygon fill=\\"#353636\\" transform=\\"rotate(45 12 12)\\" points=\\"16.5 11.4518414 12.5847458 11.4518414 12.5847458 7.5 11.4152542 7.5 11.4152542 11.4518414 7.5 11.4518414 7.5 12.5226629 11.4152542 12.5226629 11.4152542 16.5 12.5847458 16.5 12.5847458 12.5226629 16.5 12.5226629\\"></polygon></g></svg></div></div>
</div><!--SEARCH-->
    <div class=\\"hig__global-nav__top-nav__spacer\\"></div>
    <!--PROJECT_ACCOUNT_SWITCHER-->
    <div class=\\"hig__global-nav__top-nav__group hig__global-nav__top-nav__group--right\\">
    <!--SHORTCUT-->
    </div>
    <!--PROFILE-->
</div><!--TOPNAV-->
        <!--SUBNAV-->
        <div class=\\"hig__global-nav__slot\\">
            <!--SLOT-->
        </div>
    </div>
</div>"
`;

exports[`<TopNav> renders a topnav 1`] = `
"<div class=\\"hig__global-nav\\">
    <!--SIDENAV-->
    <div class=\\"hig__global-nav__container\\">
        <div class=\\"hig__global-nav__top-nav\\">
    <div class=\\"hig__global-nav__top-nav__group hig__global-nav__top-nav__group--left\\">
        <div class=\\"hig__global-nav__top-nav__hamburger\\">
            <div class=\\"hig__global-nav__top-nav__hamburger__hamburgericon\\"><div class=\\"hig__icon\\"><svg width=\\"30\\" height=\\"30\\" viewBox=\\"0 0 30 30\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"hamburger\\"><g transform=\\"translate(-92 -185)\\" fill=\\"none\\" fill-rule=\\"evenodd\\"><rect x=\\"92\\" y=\\"185\\" width=\\"30\\" height=\\"30\\"></rect><path d=\\"M100,195 L115,195 L115,196 L100,196 L100,195 Z M100,200 L115,200 L115,201 L100,201 L100,200 Z M100,205 L115,205 L115,206 L100,206 L100,205 Z\\" fill=\\"#34495E\\"></path></g></svg></div></div>
            <div class=\\"hig__global-nav__top-nav__hamburger__closeicon\\"><div class=\\"hig__icon\\"><svg width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"close-hamburger\\"><title>hig_icon_xclose</title><g fill=\\"#353636\\" fill-rule=\\"evenodd\\"><rect transform=\\"rotate(45 12 12)\\" x=\\"4\\" y=\\"11\\" width=\\"16\\" height=\\"2\\"></rect><rect transform=\\"rotate(-45 12 12)\\" x=\\"4\\" y=\\"11\\" width=\\"16\\" height=\\"2\\"></rect></g></svg></div></div>
        </div>
        <div class=\\"hig__global-nav__top-nav__logo\\">
            <a href=\\"http://www.autodesk.com\\">
                <img src=\\"../../../bim-logo.png\\">
            </a>
        </div>
    </div>
    <!--SEARCH-->
    <div class=\\"hig__global-nav__top-nav__spacer\\"></div>
    <!--PROJECT_ACCOUNT_SWITCHER-->
    <div class=\\"hig__global-nav__top-nav__group hig__global-nav__top-nav__group--right\\">
    <!--SHORTCUT-->
    </div>
    <div class=\\"hig__global-nav__profile\\">
    <div class=\\"hig__global-nav__top-nav__group hig__global-nav__top-nav__group--right\\">
        <!--FLYOUT-->
    <div class=\\"hig__flyout\\">
    <div class=\\"hig__global-nav__top-nav__profile__profile-image\\">
    <img class=\\"hig__global-nav__top-nav__profile__profile-image__image\\" src=\\"../../../bim-logo.png\\">
</div><!--TARGET-->
    <div class=\\"hig__flyout__container\\">
        <div class=\\"hig__flyout__chevron\\"></div>
        <div class=\\"hig__flyout__panel\\">
            <div class=\\"src__components__global-nav__top-nav__profile__profile-flyout-content\\">
    <div class=\\"src__components__global-nav__top-nav__profile__profile-flyout-content__name\\">name</div>
    <div class=\\"src__components__global-nav__top-nav__profile__profile-flyout-content__email\\">email@example.com</div>
    <div class=\\"src__components__global-nav__top-nav__profile__profile-flyout-content__links\\">
        <a href=\\"#\\" class=\\"hig__button hig__button--secondary hig__button--small\\" title=\\"Sign Out\\">
    
    <span class=\\"hig__button__title\\">Sign Out</span>
</a><!--SIGN_OUT_BUTTON-->
        <a href=\\"https://www.autodesk.com\\" class=\\"hig__button hig__button--secondary hig__button--small\\" title=\\"Profile Settings\\">
    
    <span class=\\"hig__button__title\\">Profile Settings</span>
</a><!--SETTINGS_LINK-->
    </div>
</div><!--SLOT-->
        </div>
    </div>
</div></div>
</div><!--PROFILE-->
</div><!--TOPNAV-->
        <!--SUBNAV-->
        <div class=\\"hig__global-nav__slot\\">
            <!--SLOT-->
        </div>
    </div>
</div>"
`;
